### PR TITLE
tests: add `integration2` for tools integration tests

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve?

Issue Number: ref #56556

Problem Summary:

Please check #56556.

### What changed and how does it work?

This is the first try of #56556. Rather than updating `tests/integrationtest`, currently I added `integrationtest2` so that the existing test scripts/test cases could be stable enough.

In the future, I will try to merge `integrationtest2` into `integrationtest` when this is stable and mature enough.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- NA

Documentation

- NA

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest2/download_integration_test_binaries.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#! /usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (bb7133) [reply tests/integrationtest2/download_integration_test_binaries.sh:1]: It is a copy of https://github.com/pingcap/tidb/blob/master/br/tests/download_integration_test_binaries.sh 

This is used mainly by CI, but I want it available for local env, too.

Also, this script guarantees we always use latest builds for all components, which is hard to archive for TiUP nightly.

[the remark this answers is not in the document]
Comment (Defined2014) [reply tests/integrationtest2/download_integration_test_binaries.sh:1]: But this shell always downloads centos7 image which can't work under MacOS env.

Using tiup means that users can manually set up the test environment through tiup.

File: tests/integrationtest2/t/dumpling_import_integration.test [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Test Lightning and AutoRandom
+
+CREATE TABLE t2(c BIGINT AUTO_RANDOM PRIMARY KEY, a INT, b INT);
+INSERT INTO t2(a, b) VALUES (1, 1), (2, 2), (3, 3);
+SELECT (c & 31) as inc FROM t2 ORDER BY inc;
+
+--dump_and_import t2 AS tt2
```
[the remark this answers is not in the document]
Comment (bb7133) [reply tests/integrationtest2/t/dumpling_import_integration.test:7]: I want it to provide an easiest way to performance 'backup' and 'restore' without any detail(for example, the syntax of BACKUP/RESTORE, the arguments of BR binary, local path of backup data...).

For integration tests with more details, many other ways like our internal testing framework can be used.

But still, maybe we can provide two new commands for `dump` and `import`.

File: tests/integrationtest2/t/br_integration.test [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Test BR and AutoIncrement
+
+CREATE TABLE t1 (a INT PRIMARY KEY NONCLUSTERED AUTO_INCREMENT, b INT) AUTO_ID_CACHE = 100;
+INSERT INTO t1 (b) VALUES (1), (2), (3);
+SHOW TABLE t1 NEXT_ROW_ID;
+
+--backup_and_restore t1 AS tt1
```
[the remark this answers is not in the document]
Comment (bb7133) [reply tests/integrationtest2/t/br_integration.test:7]: Ditto as `dump` and `import`.

**Review by hawkingrei — APPROVED** [review]

File: tests/integrationtest2/r/br_integration.result [hunk 4]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE t1 (a INT PRIMARY KEY NONCLUSTERED AUTO_INCREMENT, b INT) AUTO_ID_CACHE = 100;
+INSERT INTO t1 (b) VALUES (1), (2), (3);
+SHOW TABLE t1 NEXT_ROW_ID;
+DB_NAME	TABLE_NAME	COLUMN_NAME	NEXT_GLOBAL_ROW_ID	ID_TYPE
+br_integration	t1	_tidb_rowid	101	_TIDB_ROWID
+br_integration	t1	_tidb_rowid	1	AUTO_INCREMENT
+Destination	Size	BackupTS	Queue Time	Execution Time
+/tmp/t1_550838d3-0672-45d9-8e57-894755fe0478	3695	453145470351704099	2024-10-11 10:13:06	2024-10-11 10:13:06
```
[the remark this answers is not in the document]
Comment (bb7133) [reply tests/integrationtest2/r/br_integration.result]: I used 'enableResultLog' to control it, now it's gone. Thanks!

https://github.com/pingcap/mysql-tester/pull/133

File: tests/integrationtest2/download_integration_test_binaries.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (bb7133) [reply tests/integrationtest2/download_integration_test_binaries.sh:1]: I'll refine it later to make it work in MacOS.

**Review by Defined2014 — APPROVED** [review]

File: tests/integrationtest2/download_integration_test_binaries.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Defined2014) [reply tests/integrationtest2/download_integration_test_binaries.sh:1]: And better add `third_bin` path into `.gitignore` file

**Pull request merged**